### DISCUSSION TITLE
RadGrad2 for ICS 111

### DISCUSSION THREAD

**Discussion by philipmjohnson** [radgrad/radgrad2#486] [Ideas]
Let's consider what would be required for RadGrad2 to be integrated into the introductory programming course (ICS 111), instead of ICS 211 as we do now. 

The major benefit of this change is improved support for one of the primary goals of RadGrad (support increased engagement by women and underrepresented groups). Students have commented in the past that by the time of ICS 211, we have already "lost" some students who took ICS 111 as a way of "checking out" computer science but did not find what they were looking for. 

The major challenge is that we will need to support "pruning" of user accounts. Currently, we don't really have a good mechanism to delete users. This is not as problematic under the assumption that every user is already "pre-qualified" as an ICS major.  But if we introduce RadGrad into ICS 111, then we will need to assume that roughly half of the users that we register in RadGrad will not go on in computer science, and we need a way to detect their inactivity and delete them. 

Introducing RadGrad into ICS 111 also creates an opportunity to see if the presence of RadGrad can positively influence any students to continue on in computer science. We could do a user study after a year or so to see if any students cite RadGrad as a reason for continuing on in CS.  Currently, this makes less sense because students are already mostly committed by the time they see the system.

If we go down this path, then we could contact the instructor for ICS 111 (currently Scott) and ask if he could work with us to integrate RadGrad into the ICS 111 curriculum.  Already, he has tried to provide curriculum involving the "broader meaning and application of computer science", so I think he would be very receptive to RadGrad, which could provide a more personalized experience of this curriculum for students. 

My initial idea for pruning users is to implement a kind of "GC" for RadGrad, where the admin could manually invoke a task that:

  1. Sweeps the DB and checks to see which users have not used RadGrad, nor have taken any CS courses, for at least two years. 
  2. Shows a list of those users (and when they last visited/had courses) to the admin. The admin can de-select any students that they think should be retained.
  3. Asks the admin to confirm that they have made a recent backup of the system in case this process goes awry.
  4. Upon confirm, the task goes through and deletes that account and all related information for all specified users.  (Let's remember to also delete the Cloudinary image associated with that user, if one has been specified.)

If a student leaves CS for two years and later starts up again, we can just register them again.  They will have to start from scratch, but that's not a huge problem. 

Your thoughts?

**Comment by cammoore** [comment]
I'm teaching 111 in the fall.

**Reply by philipmjohnson** [reply]
Well that's convenient!!!!!
